test/e2e/nodeportlocal_test.go

```diff
@@ -71,30 +71,42 @@ func getNPLAnnotation(t *testing.T, data *TestData, r *require.Assertions, testP
 	var testPodIP *PodIPs
 	var found bool
 
-	_, err := data.podWaitFor(defaultTimeout, testPodName, testNamespace, func(pod *corev1.Pod) (bool, error) {
-		var err error
-		if pod.Status.Phase != corev1.PodRunning {
-			return false, nil
-		}
+	var err error
+	maxRetries := 0
+	if testOptions.enableAntreaIPAM {
+		maxRetries = 4
```

Comment:
Why is this change needed? With AntreaIPAM enabled, it will take longer time for other functions to work?
Even it's really needed, why doesn't it just increase the timeout, instead of retrying the "retry" function?

Reply:
Removed this and re-run test to check if Pods still need more time to be ready when enable AntreaIPAM.
In early phase, I detected the Pod need more time to be ready, thus I added this code.

Reply:
Reduced the timeout and added comment here.
This is mandatory for AntreaIPAM since there are agent restart cases.
```
		// If enableAntreaIPAM and agent restarted, we'll get error "http2: client connection lost" when getting Pod annotations.
		// Add more retries and reduce the timeout to handle this case.
```

Reply:
I feel the proper way to handle this is to wait for antrea-agent setting up network first, then continue the test, otherwise this could hide other issues. But anyway it doesn't affect the normal code path so should be fine for now.